src/routes/index.ts

```diff
@@ -0,0 +1,19 @@
+import { RouterBuilder } from '../utils/router-builder';
+import { getAll, createPostSchema, create, getOnePostSchema, getOne } from './posts';
+import { votesSchema, upvote, downvote } from './votes';
+
+const builder = new RouterBuilder();
+
+builder.get('/posts', getAll);
+
+builder.get('/posts/:post_id', getOne, getOnePostSchema);
+
+builder.post('/posts', create, createPostSchema);
+
+builder.put('/posts/:post_id/upvote', upvote, votesSchema);
+builder.put('/upvote/:post_id', upvote, votesSchema);
+
+builder.put('/posts/:post_id/downvote', downvote, votesSchema);
+builder.put('/downvote/:post_id', downvote, votesSchema);
```

Comment:
I felt like the `/posts/:post_id/upvote` and equivalent downvote route adhered more closely to the REST ideals since they represent actions being cast upon an entity, in this case a post.

Reply:
👍